build_tools/setup_helpers/extension.py

```diff
@@ -132,11 +132,60 @@ def get_ext_modules(debug=False):
             extra_objects=_get_extra_objects(),
             extra_link_args=_get_ela(debug),
         ),
+        _get_transducer_module(),
     ]
 
 
 class BuildExtension(TorchBuildExtension):
     def build_extension(self, ext):
         if ext.name == _EXT_NAME and _BUILD_SOX:
             _build_third_party()
+        if ext.name == _TRANSDUCER_NAME:
+            _build_transducer()
         super().build_extension(ext)
+
+
+_TRANSDUCER_NAME = '_warp_transducer'
+_TP_TRANSDUCER_BASE_DIR = _ROOT_DIR / 'third_party' / 'warp_transducer'
+
+
+def _build_transducer():
+    build_dir = str(_TP_TRANSDUCER_BASE_DIR / 'submodule' / 'build')
+    os.makedirs(build_dir, exist_ok=True)
+    subprocess.run(
+        args=['cmake', str(_TP_TRANSDUCER_BASE_DIR), '-DWITH_OMP=OFF'],
+        cwd=build_dir,
+        check=True,
+    )
+    subprocess.run(
+        args=['cmake', '--build', '.'],
+        cwd=build_dir,
+        check=True,
+    )
```

Comment:
There should be only one pair of `cmake .` and `cmake --build` calls, which in turn will call into sox build process and transducer build process. (possibly parallel),

Please introduce top level `third_party` directory and put the existing third party libraries in a dedicated directory.

```
third_party/libsox
third_party/transducer
```

Then in `third_party/CMakeLists.txt`, do 

```cmake
add_subdirectory(libsox)
add_subdirectory(transducer)
```

Then when calling `cmake`, designate the top-level `third_party` directory as the source directory and let `cmake` figure out the rest.